modules/apps/portal-security/portal-security-content-security-policy/src/main/java/com/liferay/portal/security/content/security/policy/internal/ContentSecurityPolicyHTMLRewriterImpl.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.security.content.security.policy.internal;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.content.security.policy.ContentSecurityPolicyHTMLRewriter;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Attribute;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+@Component(service = ContentSecurityPolicyHTMLRewriter.class)
+public class ContentSecurityPolicyHTMLRewriterImpl
+	implements ContentSecurityPolicyHTMLRewriter {
+
+	@Override
+	public String rewriteInlineEventHandlers(String html, String nonce) {
+		Document document = Jsoup.parse(html);
```

Comment:
Jsoup doesn't throw exceptions because it's a HTML parser and works like a browser (it does what it can).

According to the docs:

> jsoup implements the WHATWG HTML specification, and parses HTML to the same DOM as modern browsers do.
>
>    • parse HTML from a URL, file, or string
>    • find and extract data, using DOM traversal or CSS selectors
>    • manipulate the HTML elements, attributes, and text
>    • clean user-submitted content against a safelist, to prevent XSS
>    • output tidy HTML 
>
> jsoup is designed to deal with all varieties of HTML found in the wild; from pristine and validating, to invalid tag-soup; jsoup will create a sensible parse tree.

(source: https://javadoc.io/doc/org.jsoup/jsoup/1.15.3/index.html)